### DISCUSSION TITLE
Verifying that certificate and key match

### DISCUSSION THREAD

**Discussion by svend** [smallstep/certificates#969] [Q&A]
Is it possible to check that an x509 certificate and key match using the `step` CLI? With OpenSSL, you [check if the modulus strings match](https://www.ibm.com/support/pages/how-verify-if-private-key-matches-certificate).

```
openssl x509 -noout -modulus -in cert.crt | openssl md5
openssl rsa -noout -modulus -in privkey.txt | openssl md5
```

**Comment by maraino** [comment]
Hi @svend, there are different options to do that, the one that would be closer to what you show is to print the fingerprint of the public key:

```console
➜ step crypto key fingerprint cert.crt
SHA256:Voe/KK+Cr6aPW6z6NuQUncybIpZXSDuSKi6KQlWS5Og=
➜ step crypto key fingerprint cert.key
SHA256:Voe/KK+Cr6aPW6z6NuQUncybIpZXSDuSKi6KQlWS5Og=
```
